### PR TITLE
chore: rm down migrations

### PR DESCRIPTION
**Pull request by markphelps**
Closes: FLI-236

Removes all down migrations as we never exposed the ability to `migrate down` and doing so would likely open us up to potential unwanted side effects (ie migrate down, then migrate back up potentially not idempotent)

In lieu of supporting handling dirty migrations, our approach will be to inform users that databases should always be backed up before performing a migration required by a new version and users will need to fix the dirty state manually before attempting to re-migrate or use a new copy of the database.

This also switches our tests and the `import --drop` command to use the `Drop` method from `golang-migrate` instead of migrating down all the way like we were doing previously.

Interestingly I ran into an issue with SQLite and the `Drop` functionality where it failed due to constraints:

https://github.com/flipt-io/flipt/actions/runs/4557185271/jobs/8038540994#step:6:29

```
If the foreign key constraints are enabled and you perform the DROP TABLE statement, before SQLite performs the implicit DELETE statement, it carries the foreign key constraints check. If a foreign key constraint violation occurs, SQLite issues an error message and will not drop the table.
```

Therefore, I had to add a special check in our wrapped `Drop` method which disables these foreign key constraints for SQlite only before dropping.

### CODE REVIEW

File: internal/storage/sql/testing/testing.go [hunk 1]
```diff
@@ -210,6 +194,44 @@ func Open() (*Database, error) {
...
+		dr, err = pg.WithInstance(db, &pg.Config{})
+	case fliptsql.CockroachDB:
+		dr, err = cockroachdb.WithInstance(db, &cockroachdb.Config{})
+	case fliptsql.MySQL:
+		dr, err = ms.WithInstance(db, &ms.Config{})
+
+		// https://stackoverflow.com/questions/5452760/how-to-truncate-a-foreign-key-constrained-table
+		if _, err := db.Exec("SET FOREIGN_KEY_CHECKS = 0;"); err != nil {
+			return nil, fmt.Errorf("disabling foreign key checks: %w", err)
+		}
```
Comment (GeorgeMac) [inline internal/storage/sql/testing/testing.go:216]: It seems we're doing the same thing with MySQL that we do with SQLite here. Is that correct?

If so, should we now move this into the `Drop` command in the migrator alongside SQLite?

**Review by GeorgeMac — COMMENTED** [review]
Change looks great. Just one question r.e. moving MySQL foreign key constraint toggling into `Down` alongside `SQLite`.

**Review by GeorgeMac — APPROVED** [review]
One comment r.e. re-enabling foreign key constraints.
Otherwise, looks good.

File: internal/storage/sql/migrator.go [hunk 2]
```diff
@@ -126,14 +129,26 @@ func (m *Migrator) Up(force bool) error {
...
+	case SQLite:
+		// disable foreign keys for sqlite to avoid errors when dropping tables
+		// https://www.sqlite.org/foreignkeys.html#fk_enable
+		// we dont need to worry about re-enabling them since we're dropping the db
+		// and the connection will be closed
+		_, _ = m.db.Exec("PRAGMA foreign_keys = OFF")
+	case MySQL:
+		// https://stackoverflow.com/questions/5452760/how-to-truncate-a-foreign-key-constrained-table
+		_, _ = m.db.Exec("SET FOREIGN_KEY_CHECKS = 0;")
+	}
```
[the remark this answers is not in the document]
Comment (markphelps) [reply internal/storage/sql/migrator.go:146]: i think this change is only applied at a per connection level, so the next connection will have it enabled by default

**Pull request merged**